src/core_plugins/timelion/server/series_functions/es/lib/agg_response_to_series_list.js

```diff
@@ -6,8 +6,16 @@ export function timeBucketsToPairs(buckets) {
   _.each(buckets, function (bucket) {
     _.forOwn(bucket, function (val, key) {
       if (_.isPlainObject(val)) {
-        series[key] = series[key] || [];
-        series[key].push(val.value);
+        if (val.values) {
+          _.forOwn(val.values, function (bucketValue, bucketKey) {
```

Comment:
We should filter here on `NaN` values. These arise when there is no data in the time-range of the bucket. They get propagated to the client, and the code breaks on the front-end (https://github.com/thomasneirynck/kibana/blob/e274f118f720386c551d3941668387f686e3d52c/src/core_plugins/timelion/public/panels/timechart/schema.js#L168-L168).

```
TypeError: y.toFixed is not a function
    at setLegendNumbers
```